synapse/rest/admin/users.py

```diff
@@ -90,7 +90,8 @@ async def on_GET(self, request):
 
         start = parse_integer(request, "from", default=0)
         limit = parse_integer(request, "limit", default=100)
-        name = parse_string(request, "name", default=None)
+        user_id = parse_string(request, "user_id", default=None)
+        name = parse_string(request, "name", default=user_id)
```

Comment:
This still doesn't address the concern I mentioned. Now `user_id` will return results from both `displayname` and `name` columns. This breaks backwards compatibility.

What I'm asking for is:

* If `user_id` is provided, results are filtered according to the `name` column only.
* If `name` is provided, results are filtered according to the `name` and `displayname` columns.
* If both `user_id` and `name` are provided, results are filtered according to the `name` column only.